statsmodels/tsa/automatic/sarimax.py

```diff
@@ -0,0 +1,128 @@
+        for p in range(max_order[0]):
+            for q in range(max_order[1]):
+                # fit  the model
+                mod = sm.tsa.statespace.SARIMAX(endog, order=(p, d, q), **spec)
+                res = mod.fit(disp=False)
+                aic_matrix[p, q] = res.aic
+                # print(res.aic)
+        # print(aic_matrix)
+        min_aic = aic_matrix.min()
+        # print(min_aic)
+        p, q = np.where(aic_matrix == min_aic)
+        return p, q
+    else:
+        """stepwise algorithm for auto order"""
+        aic_vals = np.zeros(4)
+        order_init = {
+                        0: [2, 2],
+                        1: [0, 0],
+                        2: [1, 0],
+                        3: [0, 1]
+                    }
+        for model in range(4):
+            mod = sm.tsa.statespace.SARIMAX(
+                endog, order=(order_init[model][0], d, order_init[model][1]),
+                **spec)
+            res = mod.fit(disp=False)
+            aic_vals[model] = res.aic
+        print(aic_vals)
+        min_aic = aic_vals.min()
+        model = int(np.where(aic_vals == min_aic)[0])
+        p, q = order_init[model][0], order_init[model][1]
+        new_p, new_q = p, q
+        # only p varies by +1
+        if(p + 1 <= max_order[0]):
+            mod = sm.tsa.statespace.SARIMAX(endog, order=(p + 1, d, q), **spec)
+            res = mod.fit(disp=False)
+            # print(res.aic)
+            if (res.aic < min_aic):
+                new_p = p + 1
+                min_aic = res.aic
+        # only p varies by -1
+        if(p - 1 >= 0):
+            mod = sm.tsa.statespace.SARIMAX(endog, order=(p - 1, d, q), **spec)
+            res = mod.fit(disp=False)
+            # print(res.aic)
+            if (res.aic < min_aic):
+                new_p = p - 1
+                min_aic = res.aic
+        # only q varies by +1
+        if(q + 1 <= max_order[1]):
+            mod = sm.tsa.statespace.SARIMAX(endog, order=(p, d, q + 1), **spec)
+            res = mod.fit(disp=False)
+            # print(res.aic)
+            if (res.aic < min_aic):
+                new_q = q + 1
+                min_aic = res.aic
+        # only q varies by -1
+        if(q - 1 >= 0):
+            mod = sm.tsa.statespace.SARIMAX(endog, order=(p, d, q - 1), **spec)
+            res = mod.fit(disp=False)
+            # print(res.aic)
+            if (res.aic < min_aic):
+                new_q = q - 1
+                min_aic = res.aic
+        # both p and q vary by +1
+        if(p + 1 <= max_order[0]) and (q + 1 <= max_order[1]):
+            mod = sm.tsa.statespace.SARIMAX(endog,
+                                            order=(p + 1, d, q + 1), **spec)
+            res = mod.fit(disp=False)
+            # print(res.aic)
+            if (res.aic < min_aic):
+                new_p = p + 1
+                new_q = q + 1
+                min_aic = res.aic
+        # both p and q vary by -1
+        if(q - 1 >= 0) and (q - 1 >= 0):
```

Comment:
One of these should be `p - 1` I think>

Reply:
yes, Thanks for pointing out, I'll push the change in my next commit.

Reply:
It's often best to make commits as small as possible and avoid big commits with a bunch of changes in them. One benefit is that then the commit messages can be more informative. So, for example, I would make that change in a single commit titled something like: "BUG: order p=-1 sometimes considered".

Reply:
Thanks, @ChadFulton for letting me know about it. I have pushed the commit. I'll write the Forecast classes today and will let you know.